independent-projects/bootstrap/gradle-resolver/src/main/java/io/quarkus/bootstrap/resolver/model/impl/WorkspaceImpl.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.bootstrap.resolver.model.impl;
+
+import io.quarkus.bootstrap.resolver.model.ArtifactCoords;
+import io.quarkus.bootstrap.resolver.model.Workspace;
+import io.quarkus.bootstrap.resolver.model.WorkspaceModule;
+import java.io.Serializable;
+import java.util.Set;
+
+public class WorkspaceImpl implements Workspace, Serializable {
+
+    public ArtifactCoords mainModuleKey;
+    public Set<WorkspaceModule> modules;
```

Comment:
WorkspaceModuleImpl does not implement hashCode/equals. It would probably be better to have a map instead here. Also given that there is getModule(key)